pkg/clickhouse/source.go

```diff
@@ -102,14 +102,56 @@ func toStringSlice(iSlice []interface{}) []string {
 	return values
 }
 
-func (s Source) WriteChunk(_ string, _ io.Reader) error {
-	// TODO
-	return errors.New("not implemented")
+func (s Source) WriteChunk(_ string, r io.Reader) error {
+	ct, err := s.ColumnTypes()
+	if err != nil {
+		return err
+	}
+	tx, err := s.db.Begin()
+	if err != nil {
+		return err
+	}
+	reader := tsv.NewReader(r)
+
+	stmt, err := prepareStatement(tx, ct)
+	if err != nil {
+		return err
+	}
+
+	for {
+		records, err := reader.Read(ct)
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return err
+		}
+		_, err = stmt.Exec(records...)
+		if err != nil {
+			return err
+		}
+	}
+	defer func(stmt *sql.Stmt) {
+		err = stmt.Close()
+	}(stmt)
+	if err = tx.Commit(); err != nil {
+		return err
+	}
+
+	return err
+}
+
+func prepareStatement(tx *sql.Tx, ct []*sql.ColumnType) (*sql.Stmt, error) {
+	query := "INSERT INTO metrics VALUES ("
+	for range ct {
+		query += "?,"
+	}
+	query = query[:len(query)-1]
+	return tx.Prepare(query)
 }
```

Comment:
Could we simplify it? Also, should we add a trailing `)`?

```suggestion
func prepareInsertMetrics(tx *sql.Tx, valuesCount int) (*sql.Stmt, error) {
	query := "INSERT INTO metrics VALUES ("
	for i := 0; i < valuesCount-1; i++ {
		query += "?,"
	}
	query += "?)"
	return tx.Prepare(query)
}
```